32 store timestamps and platform usage for each prompt

barnesrj opened the pull request:
This PR adds a createdAt timestamp and website URL to each saved prompt. This allows tracking when and where prompts were created. The changes include:

Updated Prompt interface

Util to get the current browser tab URL

Modified popup.tsx to include new fields when saving

#32 
/

barnesrj committed:
Update Prompt.ts

added a variable for the date stamp and website

barnesrj committed:
Create get-tab-url.ts

created this file to get the website url

barnesrj committed:
Create get-tab-url.ts

created file to get website URL

barnesrj committed:
Update popup.tsx

Called util to save webpage; edited newPrompt

panda commented on prompt-manager/utils/prompt-manager/utils/get-tab-url.ts:
```diff
@@ -0,0 +1,12 @@
+export const getCurrentTabUrl = (): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
+      const tab = tabs[0];
+      if (tab?.url) {
+        resolve(tab.url);
+      } else {
+        reject("No active tab URL found.");
+      }
+    });
+  });
```
maybe a duplicate? I see you have this utility function in ~utils

JonCGroberg replied:
Now exists in the hook @panda @barnesrj

JonCGroberg commented on prompt-manager/interface/Prompt.ts:
```diff
@@ -1,6 +1,8 @@
+  prompt: string;
+  tags: string[];
+  category?: string;
+  createdAt: Date;   // <-- New field for timestamp
```
Remove the comments in the future

barnesrj replied:
Done.

JonCGroberg commented on prompt-manager/interface/Prompt.ts:
```diff
@@ -1,6 +1,8 @@
+  tags: string[];
+  category?: string;
+  createdAt: Date;   // <-- New field for timestamp
+  website: string;   // <-- New field for website URL
```
Remove the comments in the future

JonCGroberg commented on prompt-manager/utils/get-tab-url.ts:
function for this exists in the useURL hook

barnesrj replied:
Deleted the redundant function and replaced with useCurrentTabUrl function

barnesrj committed:
Update Prompt.ts

removed inline comments

barnesrj committed:
Update get-tab-url.ts

commented everything out before fixing with duplicate function in useURL.hook.  Will delete completely shortly.

barnesrj committed:
Update popup.tsx

Removed redundant function; changed getCurrentTabUrl to useCurrentTabUrl

barnesrj committed:
Update popup.tsx

Deleted comment.

barnesrj committed:
Delete prompt-manager/utils/get-tab-url.ts

deleted redundant function file

barnesrj committed:
Delete prompt-manager/utils/prompt-manager/utils/get-tab-url.ts

Deleted redundant function file

barnesrj committed:
Merge branch 'dev' into 32-store-timestamps-and-platform-usage-for-each-prompt

barnesrj commented:
I believe I have resolved the comments.

panda committed:
Merge branch 'dev' into 32-store-timestamps-and-platform-usage-for-each-prompt

barnesrj committed:
Update usePrompts.test.ts

Added fields to test prompts to account for new fields in the functions

barnesrj committed:
Update PromptList.test.tsx

Updated fields in test prompts.

barnesrj committed:
Update usePrompts.test.ts

Updated prompt fields.

barnesrj committed:
Update usePrompts.test.ts

Updated prompt formats with new fields.

barnesrj committed:
Update promptUtils.test.ts

Added new prompt fields

barnesrj committed:
Update PromptList.test.tsx

Added two missing commas.

barnesrj committed:
Update PromptListItem.test.tsx

Added new fields to prompt items.

barnesrj committed:
Update Prompt.ts

commented out category for error check.

barnesrj committed:
Update Prompt.ts

Uncommented category.

barnesrj committed:
Merge branch 'dev' into 32-store-timestamps-and-platform-usage-for-each-prompt

barnesrj committed:
Merge branch 'dev' into 32-store-timestamps-and-platform-usage-for-each-prompt

barnesrj committed:
Update promptUtils.ts

made it so prompt import was in date format.

panda reviewed (approved):
This looks great! Thanks for getting this done 🥳

iadeyefa reviewed (approved):
Looks great!

barnesrj committed:
Merge branch 'dev' into 32-store-timestamps-and-platform-usage-for-each-prompt
